src/Loader.php

```diff
@@ -973,4 +976,18 @@ public static function inject_wc_settings_dependencies() {
 			}
 		}
 	}
+
+	/**
+	 * Delete woocommerce_onboarding_homepage_post_id field when the homepage is deleted
+	 *
+	 * @param int $post_id The deleted post id.
+	 */
+	public static function delete_homepage( $post_id ) {
+		$homepage_id = intval( get_option( 'woocommerce_onboarding_homepage_post_id', false ) );
+		if ( 'page' !== get_post_type( $post_id ) || $homepage_id !== $post_id ) {
```

Comment:
`if ( 'page' !== get_post_type( $post_id )` could be done first to avoid having to retrieve the option unnecessarily, which would make the logic a little simpler too.

Reply:
Thanks @becdetat. Sounds good to me.
I added this change in the commit [04e05cf](https://github.com/woocommerce/woocommerce-admin/pull/4015/commits/04e05cfc5f6227d9bb3a2f5ce75d5045968b7524)